### ISSUE TITLE
Log enhancement by renaming argv[0] to appName

### ISSUE THREAD

**Issue by niuxu18** [zeroc-ice/ice#308]
We find two log revisions between Ice-3.3.1 and Ice-3.4.0, which both update argv[0] to appName when handling return value of args.empty(). Under this guidance, we have found four missed spot in the Ice-3.7.1.

Following are the spots we recommend to modify:
1. File: cpp/src/IceGrid/IceGridRegistry.cpp, Line: 114, Code:
```
bool
RegistryService::start(int argc, char* argv[], int& status)
{   ...
   if(!args.empty())
   {
      consoleErr << argv[0] << ": too many arguments" << endl;
      usage(argv[0]);
      return false;
   }
```

2. File: cpp/src/Glacier2/Glacier2Router.cpp, Line: 112, Code:
```
bool
RouterService::start(int argc, char* argv[], int& status)
{   ...
   if(!args.empty())
   {
      consoleErr << argv[0] << ": too many arguments" << endl;
      usage(argv[0]);
      return false;
   }
```

3. File: cpp/src/IceBridge/IceBridge.cpp, Line: 647, Code:
```
bool
BridgeService::start(int argc, char* argv[], int& status)
{    ...
   if(!args.empty())
   {
      cerr << argv[0] << ": too many arguments" << endl;
      usage(argv[0]);
      return false;
   }
```

4. File: cpp/src/IceStorm/Admin.cpp, Line: 96, Code:
```
int
Client::run(int argc, char* argv[])
{    ...
   if(!args.empty())
   {
      consoleErr << argv[0] << ": too many arguments" << endl;
      usage();
      return EXIT_FAILURE;
   }
```

The two historical revisions are:
1. Version: Ice-3.3.1 vs 3.4.0, File: cpp/src/IceGrid/Client.cpp, Line: 318, Patch:
```
   if(!args.empty())
   {
-       cerr << argv[0] << ": too many arguments" << endl;
+      cerr << _appName << ": too many arguments" << endl;
        usage();
       return EXIT_FAILURE;
   }
```

2. Version: Ice-3.3.1 vs 3.4.0, File:cpp/src/IcePatch2/Calc.cpp, Line: 181, Patch:
```
   if(args.empty())
   {
-       cerr << argv[0] << ": no data directory specified" << endl;
-       usage(argv[0]);
+      cerr << appName << ": no data directory specified" << endl;
+      usage(appName);
        return EXIT_FAILURE;
   }
```

Thanks for your reading, and we are looking forward to your reply about the correctness of our finding ^^

**Comment by bernardnormier** [comment]
`_appName` or `appName()` are members of the `Ice::Application` class, which is now deprecated. Some admin tools were still using `Ice::Application` as of Ice 3.7.1; this is now fixed, see #50.

Using `argv[0]` directly is fine. For example:
https://github.com/zeroc-ice/ice/blob/3.7/cpp/src/IceGrid/Client.cpp#L259

Your first 3 examples are classes derived from `Ice::Service`; while `Ice::Service` is similar to `Ice::Application`, it does not and never provided `_appName` or `appName()`.

**Comment by niuxu18** [comment]
> `_appName` or `appName()` are members of the `Ice::Application` class, which is now deprecated. Some admin tools were still using `Ice::Application` as of Ice 3.7.1; this is now fixed, see #50.
> 
> Using `argv[0]` directly is fine. For example:
> https://github.com/zeroc-ice/ice/blob/3.7/cpp/src/IceGrid/Client.cpp#L259
> 
> Your first 3 examples are classes derived from `Ice::Service`; while `Ice::Service` is similar to `Ice::Application`, it does not and never provided `_appName` or `appName()`.

Ok, i got it, thanks ^^
